src/gcsfilesystem.cpp

```diff
@@ -144,9 +144,11 @@ StatusCode GCSFileSystem::isDirectory(const std::string& path,
         client_.ListObjects(bucket, gcs::Prefix(appendSlash(object)))) {
         if (meta) {
             *is_directory = true;
-            break;
+            return StatusCode::OK;
```

Comment:
in the log SPDLOG_LOGGER_ERROR(gcs_logger, "Path is empty -> always a directory");
the message is not clear. It would be better with "Directory {} is empty. Model versions are missing"